In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score
pd.set_option('display.max_columns', None)

In [ ]:
df = pd.read_csv('../../Data/final_csv_all_data.csv')

In [35]:
df

Year  kg_per_yr_per_capita  Production_in_tons  \
0   1961             13.760740           2813910.0   
1   1962             14.116647           2872680.0   
2   1963             13.753343           2656860.0   
3   1964             13.410462           2522230.0   
4   1965             13.526203           2649230.0   
5   1966             13.969090           2422840.0   
6   1967             13.320065           2310410.0   
7   1968             14.852797           2413380.0   
8   1969             13.959552           2430900.0   
9   1970             15.192157           2842950.0   
10  1971             14.664470           2908000.0   
11  1972             16.614815           2796510.0   
12  1973             15.980325           2837670.0   
13  1974             15.358340           2894450.0   
14  1975             14.665650           2906840.0   
15  1976             16.179398           3112150.0   
16  1977             15.992288           3049560.0   
17  1978             16.753147           3474110.0   
18  1979             16.460064           3559970.0   
19  1980             16.030003           3707470.0   
20  1981             17.737926           3881180.0   
21  1982             16.967283           4134630.0   
22  1983             17.956710           4415070.0   
23  1984             19.710659           5007810.0   
24  1985             19.594719           4960690.0   
25  1986             19.401250           5196120.0   
26  1987             21.714020           5995350.0   
27  1988             20.646446           5956750.0   
28  1989             23.089434           5777930.0   
29  1990             22.487543           5871200.0   
30  1991             21.450317           5489870.0   
31  1992             22.314730           5604860.0   
32  1993             23.159950           5939830.0   
33  1994             22.679260           5926740.0   
34  1995             22.156689           5638310.0   
35  1996             21.175478           5394540.0   
36  1997             21.130096           5421800.0   
37  1998             21.894730           5154100.0   
38  1999             21.773108           5228860.0   
39  2000             22.211346           5174470.0   
40  2001             21.645264           5424700.0   
41  2002             22.810863           5436200.0   
42  2003             24.105942           5484920.0   
43  2004             24.980116           5567400.0   
44  2005             23.413918           5406890.0   
45  2006             24.286026           5372250.0   
46  2007             22.597631           5293640.0   
47  2008             21.647917           4850980.0   
48  2009             21.478006           4703280.0   
49  2010             21.463146           4808740.0   
50  2011             21.510725           5570980.0   
51  2012             21.046587           5424750.0   
52  2013             21.064482           5525190.0   
53  2014             21.621960           5406460.0   
54  2015             21.717562           5466450.0   
55  2016             21.688387           5348160.0   
56  2017             22.080343           5475780.0   
57  2018             22.443901           5264770.0   
58  2019             22.569592           5349930.0   
59  2020             22.454920           5349930.0   

    Population (historical estimates)  Life_Expectancy_years      continent  
0                           179087280                  70.27  North America  
1                           181917808                  70.12  North America  
2                           184649872                  69.92  North America  
3                           187277376                  70.17  North America  
4                           189703280                  70.21  North America  
5                           191830976                  70.21  North America  
6                           193782432                  70.56  North America  
7                           195743424                  69.95  North America  
8      

In [ ]:
df = df[df['Code'] == 'USA']

In [ ]:
df.drop(['Entity', 'Code'], inplace=True, axis=1)
df = df.dropna().reset_index(drop=True)

In [ ]:
Production_in_tons = df['Production_in_tons']
Production_in_tons

In [ ]:
Production_in_tons
data_points = Production_in_tons

In [ ]:
percentage_increase = [(data_points[i+1] - data_points[i]) / data_points[i] * 100 for i in range(len(data_points)-1)]
average_increase = np.mean(percentage_increase)
print("Average Percentage Increase: {:.2f}%".format(average_increase))

In [ ]:
x = 5349930.0
for i in range(20):
    x = x + (x * .0125)
print(x)

In [36]:
kg_per_cap = df['kg_per_yr_per_capita']

In [38]:
kg_per_cap.tail(1)

59    22.45492
Name: kg_per_yr_per_capita, dtype: float64

In [37]:
kg_per_cap
data_points = kg_per_cap
percentage_increase = [(data_points[i+1] - data_points[i]) / data_points[i] * 100 for i in range(len(data_points)-1)]
average_increase = np.mean(percentage_increase)
print("Average Percentage Increase: {:.2f}%".format(average_increase))


Average Percentage Increase: 0.96%


In [39]:
x = 22.45492
for i in range(20):
    x = x + (x * .0096)
print(x)

27.18306040179777


# attempt at prediction with ML
so the predicted values in the next 20 years for production, consumption, and population:
6858809, 27.183, 400mil

I'm going to take these values and see what the predicted life expectany is using the ML model, and compare it to actual prediction.